src/utilcode/longfilepathwrappers.cpp

```diff
@@ -1366,12 +1367,28 @@ HRESULT LongFile::NormalizePath(SString & path)
         }
     }
 
-
-    //wcscpy_s always termintes with NULL, so we are saving the character that will be overwriiten
-    WCHAR temp = buffer[prefix.GetCount()];
-    wcscpy_s(buffer, prefix.GetCount() + 1, prefix.GetUnicode());
-    buffer[prefix.GetCount()] = temp;
-    path.CloseBuffer(ret + prefixLen);
+	SString fullpath(SString::Literal, buffer + prefixLen);
+
+	//Check if the resolved path is a UNC. By default we assume relative path to resolve to disk 
+	if (fullpath.BeginsWith(UNCPathPrefix) && prefixLen != prefix.GetCount() - 2)
+	{
+		// must initialize before calling SString::Empty() via SString::Delete
+		SString::Startup();
+
+		//Remove the leading '\\' from the UNC path to be replaced with UNCExtendedPathPrefix
+		fullpath.Delete(fullpath.Begin(), 2);
+		fullpath.Insert(fullpath.Begin(), UNCExtendedPathPrefix);
```

Comment:
More efficient to combine `fullpath.Delete` and `fullpath.Insert` with `fullpath.Replace`.